src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c

```diff
@@ -1078,7 +1086,7 @@ int xocl_hwmon_sdm_init(struct xocl_dev *xdev)
 	(void) xocl_hwmon_sdm_init_sysfs(xdev, XCL_SDR_POWER);
 	(void) xocl_hwmon_sdm_init_sysfs(xdev, XCL_SDR_VOLTAGE);
 
-	return ret;
+	return 0;
```

Comment:
why are you always returning success?

Reply:
During reset, xocl_subdev_create() at line#1079 (new) returns -EEXIST (-17), and code already parse this error in following code. So, it should n't return this error code at the end, hence returning 0.

`

Reply:
Thanks @rajkumar-xilinx . Got it.